Error while checking build/O2/fullCI for 1b91241c88cd7ce425b53f2280f3c48f4666db43 at 2023-02-22 16:54:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/9704-slc8_x86-64/0/Framework/Core/src/DataProcessingDevice.cxx:128:95: error: 'mDeviceContext' was not declared in this scope; did you mean 'DeviceContext'?
/sw/SOURCES/O2/9704-slc8_x86-64/0/Framework/Core/src/DataProcessingDevice.cxx:135:25: error: 'deviceContext' was not declared in this scope; did you mean 'DeviceContext'?
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9704/1b91241c88cd7ce425b53f2280f3c48f4666db43/build_O2_fullCI/pretty.html).

--- a/Framework/Core/include/Framework/DataProcessingDevice.h
+++ b/Framework/Core/include/Framework/DataProcessingDevice.h
@@ -119,6 +119,7 @@ class DataProcessingDevice : public fair::mq::Device
   /// Handle to wake up the main loop from other threads
   /// e.g. when FairMQ notifies some callback in an asynchronous way
   uv_async_t* mAwakeHandle = nullptr;
+  int64_t mCleanupCount = -1;
 };
 
 } // namespace o2::framework

--- a/Framework/Core/src/DataProcessingDevice.cxx
+++ b/Framework/Core/src/DataProcessingDevice.cxx
@@ -125,6 +125,27 @@ DataProcessingDevice::DataProcessingDevice(RunningDeviceRef running, ServiceRegi
     mServiceRegistry{registry},
     mProcessingPolicies{policies}
 {
+  GetConfig()->Subscribe<std::string>("dpl", [&cleanupCount = mCleanupCount, &registry = mServiceRegistry](const std::string& key, std::string value) {
+    if (key == "cleanup") {
+      int64_t newCleanupCount = std::stoll(value);
+      if (newCleanupCount <= cleanupCount) {
+        return;
+      }
+      cleanupCount = newCleanupCount;
+      auto ref = ServiceRegistryRef{registry, ServiceRegistry::globalDeviceSalt()};
+      auto& deviceState = ref.get<DeviceState>();
+      for (auto& info : deviceState.inputChannelInfos) {
+        FairMQParts parts;
+        while (info.channel->Receive(parts, 0)) {
+          LOGP(debug, "Dropping {} parts", parts.Size());
+          if (parts.Size() == 0) {
+            break;
+          }
+        }
+      }
+    }
+  });
+
   std::function<void(const fair::mq::State)> stateWatcher = [this, &registry = mServiceRegistry](const fair::mq::State state) -> void {
     auto ref = ServiceRegistryRef{registry, ServiceRegistry::globalDeviceSalt()};
     auto& deviceState = ref.get<DeviceState>();

